### ISSUE TITLE
Implement "Search Files"

### ISSUE THREAD

**Issue by amourzenkov-sc** [OpusCapita/filemanager#150]
Current behaviour does not allow files/dirs searching.

1.    **Search for files according to file/dir name:**

- [x] File/dir name containing a given substring.
- [x] Either recursive or not-recursive.
- [x] Either case-sensitive or case-insensitive.
- [x] Either dirs, or files, or both files and dirs.
- [ ] Either skip archives or search in archives.
- [ ] Allow matching with patterns the shell uses.
- [ ] Interpret the substing as regular expression.

2. **Search for files according to file content:**

- [x] File content having a given substring.
- [ ] Either whole words only or any substring.
- [ ] Either case-sensitive or case-insensitive.
- [ ] Interpret the substing as regular expression.

**Comment by kvolkovich-sc** [comment]
### Request

```
findContentRegEx

findNameRecursive
findNameType
findNameRegEx
```

**Comment by amourzenkov-sc** [comment]
Implementation of **fileContentSubstring** is postponed due to its complexity:
1. Node.js streams for any file size handling.
2. Mechanism of differentiation binary files from text files.
3. Utility for reading a content of pdf/doc/odc/... files.

**Comment by amourzenkov-sc** [comment]
Sub-issues:
* #185
* #186 (depends on #185 completion)
* #187
* #188 (depends on #187 completion)
